tests/fast-integration/test/k8s-test.js

```diff
@@ -55,71 +76,91 @@ describe("Commerce Mock tests", function () {
-        [...commerceObjs, ...appConnectorObjs].map((obj) =>
-          k8sDynamicApi.create(obj)
-        )
-      );
+      await Promise.all(commerceObjs.map((obj) => k8sDynamicApi.create(obj)));
     } catch (err) {
       expect(err.body.message).to.be.empty;
     }
 
-    await sleep(5000); // TODO: add waiting for virtualservice
-
-    console.log("Waiting for virtual service to be ready...");
     let virtualservice;
     try {
-      virtualservice = await k8sCRDApi.listNamespacedCustomObject(
-        "networking.istio.io",
-        "v1beta1",
-        "mocks",
-        "virtualservices",
-        "true",
-        undefined,
-        undefined,
-        "apirule.gateway.kyma-project.io/v1alpha1=commerce-mock.mocks"
+      console.log("Waiting for virtual service to be ready...");
+      virtualservice = await retryPromise(
+        async () => {
+          return k8sCRDApi
+            .listNamespacedCustomObject(
+              "networking.istio.io",
+              "v1beta1",
+              "mocks",
+              "virtualservices",
+              "true",
+              undefined,
+              undefined,
+              "apirule.gateway.kyma-project.io/v1alpha1=commerce-mock.mocks"
+            )
+            .then((res) => {
+              expect(res.body.items).to.have.lengthOf(1);
+              expect(res.body.items[0]).to.have.nested.property("spec.hosts");
+              expect(res.body.items[0].spec.hosts).to.have.lengthOf(1);
+              expect(res.body.items[0].spec.hosts[0]).not.to.be.empty;
+              return res;
+            });
+        },
+        10,
+        1000
       );
     } catch (err) {
-      console.error(err.body.message);
+      expect(err.body.message).to.be.empty;
     }
 
-    expect(virtualservice.body.items).to.have.lengthOf(1);
-    expect(virtualservice.body.items[0].spec.hosts).to.have.lengthOf(1);
-    expect(virtualservice.body.items[0].spec.hosts[0]).not.to.be.empty;
-
     const mockHost = virtualservice.body.items[0].spec.hosts[0];
     const host = mockHost.split(".").slice(1).join(".");
-    console.log(`Host: ${host}`);
-    console.log(`Mock host: ${mockHost}`);
-
-    await sleep(40000); // TODO: add retries for axios.get instead of sleep
+    console.log(`Host: https://${host}`);
+    console.log(`Mock host: https://${mockHost}`);
 
     let response;
     try {
-      console.log(`Calling https://${mockHost}/local/apis`);
-      response = await axios.get(`https://${mockHost}/local/apis`);
-    } catch (error) {
-      expect(error).to.be.empty;
+      response = await retryPromise(
+        async () => {
+          console.log(`Calling https://${mockHost}/local/apis`);
+          return axios.get(`https://${mockHost}/local/apis`).then((res) => {
+            expect(res.data).to.have.lengthOf(2);
+            expect(res.data[0].provider).not.to.be.empty;
+            return res;
+          });
+        },
+        20,
```

Comment:
This timeout is too short. Sometimes it is more than 1 minute (Gardener Azure). 2 minutes should be enough.